components/brave_wallet_ui/components/extension/dapp-connection-settings/change-account-button.tsx

```diff
@@ -0,0 +1,223 @@
+import {
+  useSelectedAccountQuery
+} from '../../../common/slices/api.slice.extra'
+import {
+  useGetDefaultFiatCurrencyQuery,
+  useSetSelectedAccountMutation
+} from '../../../common/slices/api.slice'
+
+// Types
+import {
+  BraveWallet
+} from '../../../constants/types'
+
+// Styled Components
+import {
+  DescriptionText,
+  NameText
+} from './dapp-connection-settings.style'
+import {
+  Row,
+  Column,
+  VerticalSpace
+} from '../../shared/style'
+
+interface Props {
+  account: BraveWallet.AccountInfo
+  getAccountsFiatValue: (account: BraveWallet.AccountInfo) => Amount
+}
+
+export const ChangeAccountButton = (props: Props) => {
+  const {
+    account,
+    getAccountsFiatValue
+  } = props
+
+  // Redux
+  const dispatch = useDispatch()
+
+  // Queries
+  const { data: selectedAccount } = useSelectedAccountQuery()
+  const [setSelectedAccount] = useSetSelectedAccountMutation()
+  const isActive =
+    account.accountId.uniqueKey === selectedAccount?.accountId.uniqueKey
+  const { data: defaultFiatCurrency } = useGetDefaultFiatCurrencyQuery()
+
+  // Selectors
+  const connectedAccounts =
+    useUnsafeWalletSelector(WalletSelectors.connectedAccounts)
+
+  // Constants
+  const selectedCoin = selectedAccount?.accountId.coin
+
+  // Memos
+  const hasPermission = React.useMemo((): boolean => {
+    return connectedAccounts.some(
+      (accountId) =>
+        accountId.uniqueKey === account.accountId.uniqueKey
+    )
+  }, [connectedAccounts, account.accountId.uniqueKey])
+
+  const buttonText = React.useMemo((): string => {
+    if (selectedCoin === BraveWallet.CoinType.SOL) {
+      return hasPermission
+        ? getLocale('braveWalletSitePermissionsRevoke')
+        : getLocale('braveWalletSitePermissionsTrust')
+    }
+    return hasPermission
+      ? isActive
+        ? getLocale('braveWalletSitePermissionsDisconnect')
+        : getLocale('braveWalletSitePermissionsSwitch')
+      : getLocale('braveWalletAddAccountConnect')
+  }, [selectedCoin, hasPermission, isActive])
+
+  const accountFiatValue = React.useMemo(() => {
+    return getAccountsFiatValue(account)
+  }, [
+    getAccountsFiatValue,
+    account
+  ])
+
+  // Methods
+  const onClickConnect = React.useCallback(() => {
+    dispatch(
+      WalletActions
+        .addSitePermission({ accountId: account.accountId })
+    )
+    if (selectedCoin !== BraveWallet.CoinType.SOL) {
+      setSelectedAccount(account.accountId)
+    }
+  }, [account.accountId, selectedCoin])
+
+  const onClickDisconnect = React.useCallback(() => {
+    dispatch(
+      WalletActions
+        .removeSitePermission({ accountId: account.accountId })
+    )
+    if (
+      connectedAccounts.length !== 0 &&
+      selectedCoin !== BraveWallet.CoinType.SOL
+    ) {
+      setSelectedAccount(account.accountId)
+    }
+  }, [connectedAccounts, account.accountId, selectedCoin])
+
+  const onClickSwitchAccount = React.useCallback(() => {
+    setSelectedAccount(account.accountId)
+  }, [account])
```

Comment:
nit: we can use `account.accountId` in the deps list for perf.

Reply:
Thank you! Updated!